src/lashup_kv.erl

```diff
@@ -333,47 +386,62 @@ handle_lashup_gm_mc_event(Payload, State) ->
   lager:debug("Unknown GM MC event: ~p", [Payload]),
   State.
 
--spec(mk_full_update_fun(Key :: key(),  RemoteMap :: riak_dt_map:dt_map(), RemoteVClock :: riak_dt_vclock:vclock())
-      -> fun(() -> kv())).
-mk_full_update_fun(Key, RemoteMap, RemoteVClock) ->
+-spec(mk_full_update_fun(Key :: key(),  RemoteMap :: riak_dt_map:dt_map(),
```

Comment:
This looks sane, except I have the same feedback around the LClock. Doing a dirty_read and then using that dirty read in a transaction is not ideal.

Dirty_reads do not even promise to return us the most recent value -- even if the operations are serialized.